src/main/java/org/traccar/protocol/ZrProtocolDecoder.java

```diff
@@ -0,0 +1,323 @@
+package org.traccar.protocol;
+
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.NetworkMessage;
+import org.traccar.Protocol;
+import org.traccar.helper.BitUtil;
+import org.traccar.model.Position;
+import org.traccar.session.DeviceSession;
+
+import java.net.SocketAddress;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+/**
+ * @author QingtaiJiang
+ * @date 2023/9/18 16:54
+ * @email qingtaij@163.com
+ */
+public class ZrProtocolDecoder extends BaseProtocolDecoder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZrProtocolDecoder.class);
+
+    private static final String FRAME_HEADER = "dddd";
+    private static final String FRAME_TAIL = "ffff";
+
+    private static final int FRAME_TYPE_HEARTBEAT = 0x0100;
+    private static final int FRAME_TYPE_DEV_INFO = 0x0101;
+    private static final int FRAME_TYPE_REPORT = 0x0102;
+    private static final int FRAME_TYPE_SER_GEN_ACK = 0x0400;
+    private static final int FRAME_TYPE_LBS_TRANSFER_LONG_LAT = 0x0103;
+    private static final int FRAME_TYPE_LBS_TRANSFER_LONG_LAT_ACK = 0x0403;
+    private static final int FRAME_TYPE_TIME_SYN = 0x0104;
+    private static final int FRAME_TYPE_TIME_SYN_ACK = 0x0404;
+    public static final int FRAME_TYPE_CFG = 0x0700;
+    private static final int FRAME_TYPE_DEV_GEN_ACK = 0x0a00;
+    public static final int FRAME_TYPE_QUERY = 0x0701;
+    private static final int FRAME_TYPE_QUERY_ACK = 0x0a01;
```

Comment:
Instead of `FRAME_TYPE_` prefix, these should have prefix `MSG_`. That's the standard convention that we use.